aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -443,6 +453,16 @@ public boolean recoverFile(String path) {
     return recoverVersion.map(version -> recoverObject(version, location.bucket())).orElse(false);
   }
 
+  /**
+   * Check for a bucket name matching -does not look at endpoint.
+   *
+   * @param bucket bucket to probe.
+   * @return true if the suffix is present
+   */
+  public static boolean checkIfS3Express(final String bucket) {
+    return bucket.endsWith(S3EXPRESS_STORE_SUFFIX);
```

Comment:
also could this also be moved to `S3URI` so people can just do `s3uri.isDirectoryBucket()`?

Reply:
Think the only thing I can think of is you have to create an S3Uri object everytime you want to check if it is a directory bucket, Is that what we want? Think it is more accessible as a static method but open to suggestions,

Reply:
I think we could do both, in S3URI:

```
public static boolean isDirectoryBucket(String bucket) {
  ...
}

public boolean isDirectoryBucket() {
  return isDirectoryBucket(this.bucket);
}
```

But it feels more natural to me to have the logic in `S3URI` rather than `S3FileIO`, because this is a method to check against the S3 naming conventions, not really specific to anything in `FileIO`. If there is a `S3Util` class we could also add the method there, but we don't have that right now, and it is probably not worth adding a new class just for this small method. `S3URI` is protected and not exposed to public, so we can move things around and refactor in the future if we want.

What do you think?

Reply:
Like that way is the best of both.